test/integration/connect/envoy/run-tests.sh

```diff
@@ -142,6 +142,15 @@ function start_consul {
         '-p=9502:8502'
       )
   fi
+  
+  license="${CONSUL_LICENSE:-}"
```

Comment:
Most Consul tests when run against Consul Enterprise automatically pick up the license from the env. However the integration tests run Consul in containers so we have to go through a little more effort to get the license passed through. In OSS the presence of the env var will not hurt anything.